datastream-server-api/src/main/java/com/linkedin/datastream/server/FlushlessEventProducerHandler.java

```diff
@@ -55,6 +55,13 @@ public void clear() {
     _callbackStatusMap.clear();
   }
 
+  /**
+   * Clear the source-partition entry from the _callbackStatusMap
+   */
+  public void clear(String source, int partition) {
+    _callbackStatusMap.remove(new SourcePartition(source, partition));
```

Comment:
Just for clarification. Will this result in the correct key getting removed from the map? Does `SourcePartition` override `getHashKey()`?

Reply:
SourcePartition extends Pair underneath and uses the default Hash() function at Object level. SourcePartition was in use as Key in the HashMap for a long time.